### DISCUSSION TITLE
Syncing meetings issue

### DISCUSSION THREAD

**Discussion by julien-paris** [code4recovery/12-step-meeting-list#449] [Q&A] [answered]
Hi Brian, 
Am sending a video because I am digging into a long lasting issue we have with syncing meetings, and am hoping you can help. 
Am reclo for CER (European English speaking meetings) and ex webmaster for paris, where I am still able to login and fiddle around with the settings. 

** The specific issue I have is that when i sync meeting from Paris, and Berlin, these cities are imported as regions, as opposed to being set under their respective countries - France and Germany.**

What i do to bypass the issue is: 
1. switch them to countries 
2. sync
3. move them back under their correct country. 
https://www.loom.com/share/b4e90b0df7b943f9982c6158c3fa115e
his is not really best practice and the problem now happens with a new region, which has multiple cities. 

Any advice? 

Thanks, 

Julien

**Comment by brianw-area24** [comment]
Hi Julien,

I have to admit, I haven't had much experience with child regions. For the site I usually work on, we just use the city as the region. However, am willing to take a closer look to see if I can come up with anything for you. Would it be possible to post the file (from Paris) that you're trying to import. Either the CSV file, or the link to the json feed from the Paris Site.

If using json feed, you could always go into imports and settings, and create a new feed for "Debug", and send us that link. Then, when we're finished looking at this, you could delete that feed again from the Imports and settings page.

I'd like to try importing the Paris meetings into my development site, to see what happens.

Thanks,
Brian W

**Reply by julien-paris** [reply]
Thanks for the follow-up Brian. 
Here is the debug link: 
https://www.aaparis.org/wp-admin/admin-ajax.php?action=meetings&key=a981ccf3e4ccc05b74880a9966d2c3ab
Thanks,
Julien

**Reply by brianw-area24** [reply]
Thanks for the link. I imported meetings into my test site, and all 11 meetings were there. They are all in a region of Paris, which is a sub-Region of France, so all this appears to work as expected.

I did see a couple things, which might be relevant, both on the https://alcoholics-anonymous.eu/meetings page
- There is 2 regions for Paris, one is a subregion of France, and the other is just a region. I think you should be able to delete the one that's just a region
- The 4 meetings that aren't being imported are all at
Location: Eglise Reformee de Paris Luxembourg
Address: 58 Rue Madame, 75006 Paris, France

I wonder if there is something in the address cache on that site, that's not allowing that location to be entered. When you try to refresh the feed to import the Paris meetings, do you get any error message at the top of the imports & settings page?

Can you create a test meeting on this site, using this exact location and Address? Go ahead and try to create a test meeting, then delete it again if it's successful.

Brian W

**Comment by julien-paris** [comment]
Hi Brian, 
"There is 2 regions for Paris, one is a subregion of France, and the other is just a region. I think you should be able to delete the one that's just a region"
Yes, that is actually one of the issues. It should be a subregion of France, only. But when I sync, the Paris
 region is created. 
To bypass this issue - which happens for Paris and Berlin - I first make paris a region. then I sync. Then i make Paris a subregion of france again. 
Currently both the paris region and subregion are visible in order to debug/test/show you the issue. 

"I wonder if there is something in the address cache on that site, that's not allowing that location to be entered. When you try to refresh the feed to import the Paris meetings, do you get any error message at the top of the imports & settings page?"

I dont get an error. 

The test:
I have a created a test meeting at said location
synced Paris meetings

The result: the meeting was not synced 

Thanks, 

Julien

**Comment by brianw-area24** [comment]
One more thing I'd like to test. I have a link to the feed from Paris, would it be possible for you to post a link to the feed on the CER site also?

**Reply by julien-paris** [reply]
Sure, but can I send it to you privately?

**Reply by brianw-area24** [reply]
*address removed

**Comment by brianw-area24** [comment, marked as answer]
Thanks for sending the other feed. I'm hoping I have a solution for you, on getting the meetings into the right Paris region, under France. Here's what I'd like you to do on the CER site:
- Go to the "meetings" page in the WordPress Dashboard
- Filter by Paris (under France) and move all these meetings to trash
- Filter by the other Paris in the Regions list, and move those meeting to the Trash
- Go into trash, and delete the Paris meetings (on the feed you sent me, this was 12 meetings)
- Go to Regions page in the Dashboard, and remove both Paris's
- Go back to Imports & Settings and click Refresh for the Paris feed

So what we're doing here is removing all the Paris meetings. Then we're removing the Paris regions. Then doing a new import, hopefully this will import the 11 meetings, in a Paris region under France, at least it did on my test site.

If it doesn't import 11 meetings for Paris, let me know, and I'll send you an additional thing to try. But I'm hoping this works.

Brian W

**Reply by julien-paris** [reply]
Hi Brian, 
That worked! 
I recorded the process in case it's helpful for anyone: 
https://www.loom.com/share/f8e95be518354da0a42fd8086440f5d3

I also tested refreshing again, and it works fine, there is no new first level Paris region created as it used to do. 

What was the issue, in case it happens again? Also, since this happens with Berlin and South West France as well, I'll dive into that too.

**Comment by brianw-area24** [comment]
Glad to hear that worked!

I'm not sure what caused the issue. I was not able to duplicate the issue on my own, however, it did happen when I imported the feed from CER, so I was able to test the suggestions I sent you, and was really hoping that would do the trick. I'm hopeful that once you do the same with Berlin and South West France, that the problem won't show up again.

Thank you for your patience while working through this, and for the service work you're providing!
Brian W
